core-model-test/framework/src/main/java/org/jboss/as/core/model/test/TestParser.java

```diff
@@ -46,44 +50,115 @@ public TestParser(TestModelType type, XMLElementReader<List<ModelNode>> reader,
         }
 
+        testParser = new TestParser(type, (XMLElementReader<List<ModelNode>>) xmlParser,
+                (XMLElementWriter<ModelMarshallingContext>) xmlParser);
 
-        try {
-            for (Namespace ns : Namespace.ALL_NAMESPACES) {
-                xmlMapper.registerRootElement(new QName(ns.getUriString(), root), testParser);
-            }
-        } catch (NoSuchFieldError e) {
-            //7.1.2 does not have the ALL_NAMESPACES field
-            xmlMapper.registerRootElement(new QName(Namespace.DOMAIN_1_0.getUriString(), root), testParser);
-            xmlMapper.registerRootElement(new QName(Namespace.DOMAIN_1_1.getUriString(), root), testParser);
-            xmlMapper.registerRootElement(new QName(Namespace.DOMAIN_1_2.getUriString(), root), testParser);
+
+        Class<?> namespaceClass = Class.forName("org.jboss.as.controller.parsing.Namespace");
+        Field allNamespacesField = namespaceClass.getDeclaredField("ALL_NAMESPACES");
+        Object[] allNamespaces = (Object[]) allNamespacesField.get(null);
+        Method getUriStringMethod = namespaceClass.getMethod("getUriString");
+
+        for (Object ns : allNamespaces) {
+
+            xmlMapper.registerRootElement(new QName(getUriStringMethod.invoke(ns).toString(), root), testParser);
         }
+
         return testParser;
     }
 
+    private static TestParser createFrom27(ExtensionRegistry registry, XMLMapper xmlMapper, TestModelType type) {
+        Stability stability = Stability.DEFAULT;
+
+        ManagementSchemas schemas;
+        if (type == TestModelType.STANDALONE) {
+            schemas = new StandaloneXmlSchemas(stability, null, Executors.newCachedThreadPool(), registry);
+        } else if (type == TestModelType.DOMAIN) {
+            schemas = new DomainXmlSchemas(stability, null, Executors.newCachedThreadPool(), registry);
+        } else if (type == TestModelType.HOST) {
+            schemas = new HostXmlSchemas(stability, "primary", RunningMode.NORMAL, false, null, Executors.newCachedThreadPool(), registry);
+        } else {
+            throw new IllegalArgumentException("Unknown type " + type);
+        }
+
+        ManagementXmlSchema schema = schemas.getCurrent();
+        TestParser testParser = new TestParser(type, schema, schema);
+
+        xmlMapper.registerRootElement(schema.getQualifiedName(), testParser);
+
+        for (ManagementXmlSchema additional : schemas.getAdditional()) {
+            xmlMapper.registerRootElement(additional.getQualifiedName(), testParser);
+        }
+
+        return testParser;
+    }
+
+
+    public static TestParser create(ExtensionRegistry registry, XMLMapper xmlMapper, TestModelType type) {
+        if (getModelMajorVersion() < 27) {
+            try {
+                return createLegacy(registry, xmlMapper, type);
+            } catch (Exception e) {
+                throw new IllegalStateException("Failed to create the parser", e);
```

Comment:
i18n

Reply:
@bstansberry This one is just a part of the core-model-test framework so I think we can stick with English only?

Reply:
ah, right. No problem.